docs/grammar/Solidity.g4

```diff
@@ -0,0 +1,489 @@
+/**
+ * Solidity is a statically typed, contract-oriented, high-level language for implementing smart contracts on the Ethereum platform.
+ */
+grammar Solidity;
+
+options { tokenVocab=SolidityLexer; }
+
+/**
+ * On top level, Solidity allows pragmas, import directives, and
+ * definitions of constracts, interfaces, libraries, structs and enums.
+ */
+sourceUnit: (
+    pragmaDirective
+    | importDirective
+    | contractDefinition
+    | interfaceDefinition
+    | libraryDefinition
+    | structDefinition
+    | enumDefinition
+)* EOF;
+
+//@doc: inline
+pragmaDirective: Pragma PragmaToken+ PragmaSemicolon;
+
+/**
+ * Import directives import identifiers from different files.
+ */
+importDirective:
+    Import (
+	    (path (As unitAlias=identifier)?)
+	    | (symbolAliases From path)
+	    | (Mul As unitAlias=identifier From path)
+	) Semicolon;
+//@doc: inline
+//@doc:name aliases
+importAliases: symbol=identifier (As alias=identifier)?;
+/**
+ * Path of a file to be imported.
+ */
+path: NonEmptyStringLiteral;
+/**
+ * List of aliases for symbols to be imported.
+ */
+symbolAliases: LBrace aliases+=importAliases (Comma aliases+=importAliases)* RBrace;
+
+/**
+ * Top-level definition of a contract.
+ */
+contractDefinition:
+	Abstract? Contract name=identifier
+	inhertianceSpecifierList?
+	LBrace contractBodyElement* RBrace;
+/**
+ * Top-level definition of an interface.
+ */
+interfaceDefinition:
+	Interface name=identifier
+	inhertianceSpecifierList?
+	LBrace contractBodyElement* RBrace;
+/**
+ * Top-level definition of a library.
+ */
+libraryDefinition: Library name=identifier LBrace contractBodyElement* RBrace;
+
+//@doc:inline
+inhertianceSpecifierList:
+    Is inheritanceSpecifiers+=inheritanceSpecifier
+    (Comma inheritanceSpecifiers+=inheritanceSpecifier)*?;
+/**
+ * Inheritance specifier for contracts and interfaces.
+ * Can optionally supply base constructor arguments.
+ */
+inheritanceSpecifier: name=userDefinedTypeName arguments=callArgumentList?;
+
+/**
+ * Declarations that can be used in contracts, interfaces and libraries.
+ *
+ * Note that interfaces and libraries may not contain state variables and constructors and libraries additionally
+ * may not contain fallback or receive functions.
```

Comment:
```suggestion
 * may not contain fallback nor receive functions.
```

Reply:
Can't libraries have constant state variables?

Reply:
True, changing the comment.